src/coprocessor/endpoint.rs

```diff
@@ -828,6 +868,38 @@ impl<E: Engine> Endpoint<E> {
             .or_else(|e| futures::future::ok(make_error_response(e))) // Stream<Resp, ()>
             .map(|item: std::result::Result<_, ()>| item.unwrap())
     }
+
+    fn read_pool_spawn_with_memory_quota_check<F>(
+        &self,
+        mut allocated_bytes: usize,
+        future: F,
+        priority: CommandPri,
+        task_id: u64,
+        metadata: TaskMetadata<'_>,
+        resource_limiter: Option<Arc<ResourceLimiter>>,
+    ) -> Result<()>
+    where
+        F: Future<Output = ()> + Send + 'static,
+    {
+        allocated_bytes += mem::size_of_val(&future);
+        let mut owned_quota = OwnedAllocated::new(self.memory_quota.clone());
+        owned_quota.alloc(allocated_bytes)?;
+        COPR_MEMORY_QUOTA
+            .in_use
+            .set(owned_quota.source().in_use() as _);
+        let fut = future.map(move |_| {
+            let in_use = owned_quota.source().in_use();
+            let allocated = owned_quota.allocated();
+            COPR_MEMORY_QUOTA
```

Comment:
NIT: Seems there is no need to update the metrics this frequently. Is it possible to just update it periodically or make it as a `MetricsCollector`.